`configure` fails when using `--with-spicy=<dir>`

Was testing some local Spicy stuff and tried using `--with-spicy` rather than just changing the submodule. But `cmake` errors:

```
$ ./configure --generator=Ninja --with-spicy=/Users/etyp/.local/spicy
<...>
CMake Error at src/spicy/spicyz/CMakeLists.txt:21 (hilti_link_libraries_in_tree):
  Unknown CMake command "hilti_link_libraries_in_tree".


-- Configuring incomplete, errors occurred!
Also, before re-running configure, consider cleaning the cache (removing the build directory) via `make distclean`
```

--- a/src/spicy/spicyz/CMakeLists.txt
+++ b/src/spicy/spicyz/CMakeLists.txt
@@ -10,16 +10,20 @@ set_target_properties(spicyz PROPERTIES CXX_EXTENSIONS OFF)
 
 target_include_directories(spicyz PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 
-if (BINARY_PACKAGING_MODE)
-    # Ensure we link against the HILTI/Spicy toolchain libraries (should be
-    # default, but isn't set here for some reason).
-    set(HAVE_TOOLCHAIN ON)
-
-    hilti_link_object_libraries_in_tree(spicyz PRIVATE)
-    spicy_link_object_libraries_in_tree(spicyz PRIVATE)
+if (SPICY_ROOT_DIR)
+    target_link_libraries(spicyz PRIVATE hilti spicy)
 else ()
-    hilti_link_libraries_in_tree(spicyz PRIVATE)
-    spicy_link_libraries_in_tree(spicyz PRIVATE)
+    if (BINARY_PACKAGING_MODE)
+        # Ensure we link against the HILTI/Spicy toolchain libraries (should be
+        # default, but isn't set here for some reason).
+        set(HAVE_TOOLCHAIN ON)
+
+        hilti_link_object_libraries_in_tree(spicyz PRIVATE)
+        spicy_link_object_libraries_in_tree(spicyz PRIVATE)
+    else ()
+        hilti_link_libraries_in_tree(spicyz PRIVATE)
+        spicy_link_libraries_in_tree(spicyz PRIVATE)
+    endif ()
 endif ()
 
 prefer_configured_spicy_include_dirs(spicyz)

--- a/src/spicy/spicyz/driver.cc
+++ b/src/spicy/spicyz/driver.cc
@@ -13,13 +13,13 @@
 
 #include <hilti/ast/declarations/type.h>
 #include <hilti/compiler/init.h>
+#include <hilti/compiler/plugin.h>
 
 #include <spicy/ast/types/unit.h>
 #include <spicy/ast/visitor.h>
 #include <spicy/autogen/config.h>
 #include <spicy/compiler/init.h>
 
-#include "compiler/plugin.h"
 #include "config.h"
 #include "glue-compiler.h"
 

